In [0]:
# numpyをインポート
import numpy as np

## 1. 指数関数, 対数関数  
### ネイピア数を底とする指数関数
ネイピア数$ e^x $の計算をする場合は，numpy.exp()を用いることができます．引数には数値だけでなく，配列を指定することも可能です．  
【例題1】xを0, 1, 2とした場合のnumpy.exp()の返り値を確認します．以下のコードを読んで実行してみましょう．

In [2]:
# x = 0, 1, 2
x = np.array([0, 1, 2])

# exp(0), exp(1), exp(2)を計算
y = np.exp(x)
print(y)

[1.         2.71828183 7.3890561 ]


- 値が1, 2.71828183, 7.3890561となっていれば成功です．  
- ネイピア数e自体はnumpy.eで呼び出すことができます．

【問題1】以下に示すシグモイド関数を実装してみましょう．  
$$
    h(x)=\frac{1}{1+e^{-x}}
$$

In [3]:
# xに1を代入
x = 1.

# シグモイド関数の実装
sigmoid = 1 / (1 + np.exp(-x))

print('h(1.0) = ', sigmoid)

h(1.0) =  0.7310585786300049


- 値が0.731....となっていれば正解です．  
- exp(x)の計算はnp.exp(x)で求められることを確認しましょう．

### 対数関数  
底がネイピア数eの対数関数はnumpy.log()によって計算することができます．こちらも引数に配列を指定することが可能です.  
【例題2】numpy.log()に配列を与えた場合の結果を確認します．以下のコードを読んで実行してみましょう．

In [4]:
# 配列0, 1, 2を生成
x = np.array([1, 2, 3])

# log(x)を計算
y = np.log(x)

print(y)

[0.         0.69314718 1.09861229]


- 答えが0, 0.693..., 1.098...となっていれば成功です．  
- 底が2の場合はnumpy.log2(), 底が10の場合はnumpy.log10()によって計算することが可能です．

【問題2】numpy.log()を用いて$$ log(e^2) $$を計算してみましょう．

In [6]:
print(np.log(np.e**2))

2.0


- 答えが2.0となっていれば正解です．

## 2. 平方根，絶対値  
### 平方根  
平方根を求めたいときはnumpy.sqrt()を用います．  
【例題1】$ \sqrt{2} $を求めます．コードを読んで実行してみましょう．

In [5]:
# √2を求める
print(np.sqrt(2))

1.4142135623730951


- 答えが1.414...となっていれば成功です．

【問題1】以下に示すL2ノルムを実装してみましょう．  
$$
    ||x||_2=\sqrt{\sum_{k=1}^{n}x_k^2}
$$

In [7]:
# xを[1, 2, 3]とする．
x = np.array([1, 2, 3])

# L2ノルムの実装
norm = np.sqrt(np.sum(x**2))

print(norm)

3.7416573867739413


*斜体テキスト*- 値が3.74...となっていれば正解です．

### 絶対値  
絶対値を求めるにはnumpy.abs()を用います．  
【例題2】numpy.abs()を用いて絶対値を求めます．コードを読んで実行してみましょう．

In [8]:
# -2.5の絶対値を求める
print(np.abs(-2.5))

2.5


- 値が2.5となっていれば成功です．

【問題】[1, -2, 3, -4]を要素として持つ配列を生成し，それらの絶対値を求めてみましょう．

In [9]:
# 配列の生成
x = np.array([1, -2, 3, -4])

# 絶対値を求める
print(np.abs(x))

[1 2 3 4]


- 値が1, 2, 3, 4となっていれば正解です．